tests/unit-tests/cart/cart.php

```diff
@@ -2009,6 +2009,57 @@ public function test_changing_tax_class_via_filter_issue_23340() {
 		remove_filter( 'woocommerce_product_variation_get_tax_class', array( $this, 'change_tax_class_filter' ) );
 	}
 
+	/**
+	 * Test rounding with fees as described in Github issue 25629.
+	 */
+	public function test_rounding_with_fees_25629() {
+		update_option( 'woocommerce_prices_include_tax', 'yes' );
+		update_option( 'woocommerce_calc_taxes', 'yes' );
+		update_option( 'woocommerce_tax_round_at_subtotal', 'yes' );
+
+		WC()->cart->empty_cart();
+		$tax_rate    = array(
+			'tax_rate_country'  => '',
+			'tax_rate_state'    => '',
+			'tax_rate'          => '10.0000',
+			'tax_rate_name'     => 'TAX10',
+			'tax_rate_priority' => '1',
+			'tax_rate_compound' => '0',
+			'tax_rate_shipping' => '1',
+			'tax_rate_order'    => '1',
+			'tax_rate_class'    => '10percent',
```

Comment:
Need to create this tax class first with `WC_Tax::create_tax_class( '10percent' );`